src/Okta.Idx.Sdk/OktaVerify/OktaVerifyAuthenticationOptions.cs

```diff
@@ -0,0 +1,253 @@
+                    };
+                }
+
+                if (challengeResponse.ContainsRemediationOption(RemediationType.SelectAuthenticatorEnroll))
+                {
+                    return new AuthenticationResponse
+                    {
+                        IdxContext = IdxContext,
+                        AuthenticationStatus = AuthenticationStatus.AwaitingAuthenticatorEnrollment,
+                        Authenticators = IdxResponseHelper.ConvertToAuthenticators(challengeResponse.Authenticators.Value),
+                    };
+                }
+
+                throw new UnexpectedRemediationException(
+                    new List<string>
+                    {
+                                    RemediationType.SelectAuthenticatorAuthenticate,
+                                    RemediationType.SelectAuthenticatorEnroll,
+                    },
+                    challengeResponse);
+            }
+            catch (Exception ex)
+            {
+                return new AuthenticationResponse
+                {
+                    AuthenticationStatus = AuthenticationStatus.Exception,
+                    MessageToUser = ex.Message,
+                };
+            }
+        }
+
+        /// <summary>
+        /// Sends a request to the challenge polling endpoint.
+        /// </summary>
+        /// <returns>OktaVerifyPollResponse.</returns>
+        /// <exception cref="ArgumentException">If the challenge authenticator is not valid.</exception>
+        public async Task<OktaVerifyPollResponse> PollOnceAsync()
+        {
+            if (ChallengePollRemediationOption.Name != RemediationType.ChallengePoll)
+            {
+                throw new ArgumentException($"Expected remediation option of type '{RemediationType.ChallengePoll}', the specified remediation option is of type {ChallengePollRemediationOption.Name}.");
+            }
+
+            IdxRequestPayload requestPayload = new IdxRequestPayload
+            {
+                StateHandle = StateHandle,
+            };
+
+            try
+            {
+                var challengeResponse = await ChallengePollRemediationOption.ProceedAsync(requestPayload);
+                bool continuePolling = challengeResponse.ContainsRemediationOption(RemediationType.ChallengePoll, out IRemediationOption challengePollRemediationOption);
+
+                if (challengeResponse.SuccessWithInteractionCode != null)
+                {
+                    TokenInfo = await challengeResponse.SuccessWithInteractionCode.ExchangeCodeAsync(IdxContext);
+                    return new OktaVerifyPollResponse
+                    {
+                        AuthenticationStatus = AuthenticationStatus.Success,
+                        TokenInfo = TokenInfo,
+                        ContinuePolling = continuePolling,
```

Comment:
If we have TokenInfo already provided, do we need the continuePolling flag?

Reply:
It is set to false here to advise the client that it can move on.